src/app/frontend/filter_dataframe.py

```diff
@@ -72,9 +74,10 @@ def filter_dataframe(df: pd.DataFrame) -> pd.DataFrame:
     - Text columns: Supports filtering based on substrings or regular expressions.
     """
     st.session_state[NCBI_DF_FILTER] = st.checkbox("Add filters")
+    filters = {}  # Dictionary to store the applied filters
 
     if st.session_state[NCBI_DF_FILTER] is False:
-        return df
+        return df, filters
```

Comment:
Rather than a return here, wouldn't it be better to capture the case when `st.session_state[NCBI_DF_FILTER] is True`, so like:

```python
if st.session_state[NCBI_DF_FILTER]: # if true
    ... # The preprocessing code stuff

return df, filters
```

essentially if `True` then further processing is done modifying the `df` and `filters`, then return the modified value at the end. If things are `False` then no modification is needed, so just return the non-modified `df` and `filters` 😄

Reply:
Didn't catch this initially! Sorry!

Reply:
I was thinking that `st.session_state[NCBI_DF_FILTER] is False` is the base condition, so do nothing and return.

If the above base condition fails, then proprocess the dataframe and show the UI for filtering this dataframe.

If I add a condition for `True` case like you mentioned in the above code snippet, then everything from `line 82` till `line 130` will be indented under this condition, which I felt was less readable.

What do you think?

Reply:
> If I add a condition for True case like you mentioned in the above code snippet, then everything from line 82 till line 130 will be indented under this condition, which I felt was less readable.

This is true, but you can collapse this easily if using an IDE. Also the double return statement that is exactly the same kinda breaks the Don't Repeat Yourself motto 😄 If one need to find the return statement quickly they would find the first one and with the giant block of code, you might miss the last one.